consensus/src/pos/mod.rs

```diff
@@ -34,36 +34,24 @@ use crate::pos::{error::ConsensusPoSError, kernel::get_kernel_output};
 
 fn check_stake_kernel_hash<P: PoSAccountingView>(
     epoch_index: EpochIndex,
-    random_seed: &H256,
+    random_seed: &PoSRandomness,
     pos_data: &PoSData,
     kernel_output: &TxOutput,
     spender_block_header: &BlockHeader,
     pos_accounting_view: &P,
 ) -> Result<(), ConsensusPoSError> {
-    let target: Uint256 = (*pos_data.bits())
+    let target: Uint256 = (*pos_data.target())
```

Comment:
The retarget algorithm is missing. With the current implementation, the block producer can put any target they want. There should be an algorithm that enforces that every target in the current block follows a given recipe.

Reply:
yes, I know, thought about adding it separately